python/pyarrow/__init__.py

```diff
@@ -151,6 +151,29 @@ def parse_git(root, **kwargs):
 
 import pyarrow.types as types
 
+# GPU support
+#
+try:
+    import pyarrow.lib_gpu as _lib_gpu
+except ImportError as _msg:
+    _msg = str(_msg)
+    if _msg.startswith('No module named') and 'lib_gpu' in _msg:
+        has_gpu_support = False
+    else: # do not silence ImportError in case of any bugs
+        raise
+else:
+    has_gpu_support = True
+
+if has_gpu_support:
+    from pyarrow.lib_gpu \
```

Comment:
I wonder if it is a good idea to keep injecting everything in the top-level namespace. Though that's what we've been doing until now... @wesm

Reply:
I have introduced pyarrow/cuda.py and cleaned up __init__.py. Failure to import pyarrow.cuda indicates the lack of cuda support.